metadata-integration/java/datahub-client/src/main/java/datahub/client/rest/RestEmitter.java

```diff
@@ -154,7 +155,12 @@ public static RestEmitter createWithDefaults() {
   @Override
   public Future<MetadataWriteResponse> emit(MetadataChangeProposalWrapper mcpw,
       Callback callback) throws IOException {
-    return emit(this.eventFormatter.convert(mcpw), callback);
+    try {
+      return emit(StringEscapeUtils.convert(mcpw), callback);
+    } catch (URISyntaxException e) {
+      log.error(e.toString());
```

Comment:
Generally we try to avoid both logging and throwing simultaneously

Reply:
Done